flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -375,6 +378,51 @@ public RunnableFuture<SnapshotResult<KeyedStateHandle>> snapshot(
             @Nonnull CheckpointStreamFactory streamFactory,
             @Nonnull CheckpointOptions checkpointOptions)
             throws Exception {
+
+        if (checkpointOptions.getCheckpointType().isSavepoint()) {
+            SnapshotType.SharingFilesStrategy sharingFilesStrategy =
+                    checkpointOptions.getCheckpointType().getSharingFilesStrategy();
+            if (sharingFilesStrategy == SnapshotType.SharingFilesStrategy.NO_SHARING) {
```

Comment:
Do you mean making the judgement before the savepoint condition ?
I'd make it just in the savepoint condition to avoid affecting normal checkpoint logic. 
I have extracted the native savepoint logic to a function and make other strategies exit firstly in the funtion.